pytools/tag.py

```diff
@@ -321,6 +323,37 @@ def tags_of_type(self, tag_t: Type[TagT]) -> FrozenSet[TagT]:
                          for tag in self.tags
                          if isinstance(tag, tag_t)})
 
+    @memoize_method
+    def tags_not_of_type(self, tag_t: Type[TagT]) -> FrozenSet[Tag]:
+        """
+        Returns *self*'s tags that are not of type *tag_t*.
+        """
+        return frozenset({tag
+                         for tag in self.tags
+                         if not isinstance(tag, tag_t)})
+
+    def __eq__(self, __o: object) -> bool:
+        new_self_tags = self.tags_not_of_type(IgnoredForEqualityTag)
+        if isinstance(__o, Taggable):
+            new_other_tags = __o.tags_not_of_type(IgnoredForEqualityTag)
+            return new_self_tags == new_other_tags
+        else:
+            return super().__eq__(__o)
+
+    def __hash__(self) -> int:
+        return hash(self.tags_not_of_type(IgnoredForEqualityTag))
+
+
+# }}}
+
+
+# {{{ IgnoredForEqualityTag
+
+class IgnoredForEqualityTag(Tag):
+    """A type of tag whose value is purely informational and should not be used
+    for equality comparison."""
```

Comment:
Be more precise here: What does "not be used for equality comparison" mean? Use words like presence, absence, subclass, equality of the actual tags.

Reply:
What do you think of [15e9ce6](https://github.com/inducer/pytools/pull/134/commits/15e9ce6fd18d0e0aaad2ffa7db9ba9b9d238b0ba)?